app/controllers/candidate_interface/personal_statement_controller.rb

```diff
@@ -34,7 +38,14 @@ def update
       @return_to = return_to_after_edit(default: candidate_interface_becoming_a_teacher_show_path)
 
       if @becoming_a_teacher_form.save(current_application)
-        redirect_to @return_to[:back_path]
+        if @becoming_a_teacher_form.blank?
+          if current_application.becoming_a_teacher_completed?
+            current_application.update!(becoming_a_teacher_completed: false)
+          end
+          redirect_to candidate_interface_application_form_path
+        else
+          redirect_to @return_to[:back_path]
+        end
```

Comment:
is there a way of doing this with less nesting of conditionals? this is quite difficult to read, and therefore to maintain.

Reply:
Good question? Any ideas? 😄

Reply:
I did originally not have the condition, and just always set `becoming_a_teacher_completed: false`, on the basis that it didn't do any harm if it was already false.

But then I realised that it could be `nil`, which makes a difference as that means there's no default answer in the "Is this section completed?" – so it needed the condition...

So method name would have to be something subtle like

```
def set_section_to_incomplete_if_completed
  if current_application.becoming_a_teacher_completed?
    current_application.update!(becoming_a_teacher_completed: false)
  end  
end
```

Does that help?

Reply:
@carlosmartinez have done this. Let me know if you can think of any other changes.

I think possibly the create and update actions and routes could be combined, as the page is identical and the logic is almost the same, but that would be a bigger refactor.